File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
class Solution {
    public int[] solution(int n, long left, long right) {
        
        int[] answer = new int[(int)right-(int)left+1];
        int index = 0;
        for(long i = left; i <= right; i++) {
            long row = i / n;
            long col = i % n;
            long max = Math.max(row,col);
            answer[index++] = (int)(max + 1); 
        }
       
        
        
        return answer;
    }
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.js
function solution(n, left, right) {
    var answer = [];
    for (let i = left; i <= right; i++) {
        const row = Math.floor(i / n);
        const col = i % n;
        const max = Math.max(row,col);
        answer.push(max+1);
    }
    return answer;
}